### DISCUSSION TITLE
Stuck in the beginning

### DISCUSSION THREAD

**Discussion by delirans** [AntonOsika/gpt-engineer#289] [General]
Hi guys, I feel terrible to ask for help with what seems like a really basic problem but I can't get pass this. I feel like an idiot already.

Here's the issue: I'm using Windows and when set my openai key using "set OPENAI_API_KEY" in the terminal and then running main.py I get this long error

> Traceback (most recent call last):
> 
>   File "C:\Users\0ot6few\VS Code\gpt-engineer\gpt_engineer\main.py", line 62, in <module>
>     app()
> 
>   File "C:\Users\ot6few\VS Code\gpt-engineer\gpt_engineer\main.py", line 43, in main
>     ai = AI(
>          ^^^
> 
>   File "C:\Users\0ot6few\AppData\Local\Programs\Python\Python311\Lib\site-packages\gpt_engineer\ai.py", line 13, in __init__
>     openai.Model.retrieve(model)
> 
>   File "C:\Users\0ot6few\AppData\Local\Programs\Python\Python311\Lib\site-packages\openai\api_resources\abstract\api_resource.py", line 20, in retrieve
>     instance.refresh(request_id=request_id, request_timeout=request_timeout)
> 
>   File "C:\Users\0ot6few\AppData\Local\Programs\Python\Python311\Lib\site-packages\openai\api_resources\abstract\api_resource.py", line 32, in refresh
>     self.request(
> 
>   File "C:\Users\0ot6few\AppData\Local\Programs\Python\Python311\Lib\site-packages\openai\openai_object.py", line 172, in request
>     requestor = api_requestor.APIRequestor(
>                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
>   File "C:\Users\0ot6few\AppData\Local\Programs\Python\Python311\Lib\site-packages\openai\api_requestor.py", line 138, in __init__
>     self.api_key = key or util.default_api_key()
>                           ^^^^^^^^^^^^^^^^^^^^^^
> 
>   File "C:\Users\0ot6few\AppData\Local\Programs\Python\Python311\Lib\site-packages\openai\util.py", line 186, in default_api_key
>     raise openai.error.AuthenticationError(
> 
> openai.error.AuthenticationError: No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. See https://platform.openai.com/account/api-keys for details.

And when I add my key straight into main.py I get this error:

> Traceback (most recent call last):
>   File "C:\Users\0ot6few\VS Code\gpt-engineer\gpt_engineer\main.py", line 16, in <module>
>     openai.api_key = "sk......"
>     ^^^^^^
> NameError: name 'openai' is not defined. Did you mean: 'open'?

So what the hell I'm I doing wrong here?

Many thanks in advance!

**Comment by patillacode** [comment]
I am pretty sure you are not running the app within the correct environment.

If you wish to reopen the issue please do following the new issue template.
